cli/integration/plugin_registry_list_integration_test.go

```diff
@@ -210,43 +210,56 @@ provider:
 `, name, version)
 }
 
-func GeneratePluginReferences(name, version, description string, platforms []string) string {
-	var s string
+func CreateCompRefNameFromPlugin(name string) string {
+	return strings.ReplaceAll(strings.ReplaceAll(name, ".", "-"), "/", "-")
+}
+
+func GeneratePluginReferences(componentName, version, description string, platforms []string) string {
+	name := CreateCompRefNameFromPlugin(componentName)
+
+	var b strings.Builder
 
-	s += fmt.Sprintf(`
+	_, _ = fmt.Fprintf(&b, `
   - name: %s
     version: %s
     componentName: %s
-`, name, version, name)
+`, name, version, componentName)
 
-	// Add labels if description or platforms are provided
-	var labels string
+	// Build pluginInfo payload
+	info := make(map[string]any)
 	if description != "" {
-		labels += fmt.Sprintf(`
-          description: %s
-`, description)
+		info["description"] = description
 	}
-
 	if len(platforms) > 0 {
-		labels += `
-          platforms:
-`
-
-		for _, platform := range platforms {
-			labels += fmt.Sprintf(`
-            - %s`, strings.TrimSpace(platform))
+		trimmed := make([]string, len(platforms))
+		for i, p := range platforms {
+			trimmed[i] = strings.TrimSpace(p)
 		}
+		info["platforms"] = trimmed
 	}
 
-	if labels != "" {
-		s += fmt.Sprintf(`
+	if len(info) > 0 {
+		jsonBytes, _ := json.MarshalIndent(info, "", "  ")
+		jsonStr := string(jsonBytes)
+
+		fmt.Fprintf(&b, `
     labels:
       - name: %s
-        value:
-%s`, list.PluginInfoKey, labels)
+        value: |
```

Comment:
This feels like I am breaking current plugins - as per `schem.json`, we are not allowed to use yaml objects but only strings for `label.value`

see https://github.com/open-component-model/open-component-model/blob/main/bindings/go/constructor/spec/v1/resources/schema-2020-12.json#L304

`label.value` ref https://github.com/open-component-model/open-component-model/blob/main/bindings/go/constructor/spec/v1/resources/schema-2020-12.json#L45

Reply:
According to old ocm, label should be an object in v2
https://github.com/open-component-model/ocm/blob/main/resources/component-descriptor-v2-schema.yaml#L29

and v3
https://github.com/open-component-model/ocm/blob/main/resources/component-descriptor-ocm-v3-schema.yaml#L44

so I will move this into new ocm as well

Reply:
Nice catch!